### DISCUSSION TITLE
Modals on modals...

### DISCUSSION THREAD

**Discussion by cfepdan** [Budibase/budibase#5324] [Ideas]
I want to have a small pop-up confirmation for an action that occurs within a modal. If I navigate to a page as a modal from within a modal, the new page replaces my existing modal...

In this specific case, it'd be helpful to have the ability to create more than one layer of modal.

![image](https://user-images.githubusercontent.com/28074021/162537047-d99a9ed4-1131-48eb-b472-37d8d3c5bf7c.png)

**Comment by cfepdan** [comment]
Fortunately what I'm trying to accomplish here is a row deletion, which does this neatly, if not highly customizably.

**Comment by melohagan** [comment]
Hi @cfepdan 

Having many nested modals is probably not ideal, but I can definitely see the use case for wanting a confirmation dialog like you've shown. 
Maybe it could be a new action: **Show Confirmation Dialog** that takes:

- title (can be left blank)
- body
- cancel text
- submit text
- submit action

And this dialog always gets shown at top level, regardless of whether you have a modal open or not.
